### PR TITLE
feat(vault): add logic to obtain/refresh token with aws auth method

### PR DESCRIPTION
**Pull request by JordanSussman**
This code should be fairly easy to modify down the road incase someone would like to add another [auth method](https://www.vaultproject.io/docs/auth) to obtain the token.

closes https://github.com/go-vela/community/issues/106

### CODE REVIEW

File: cmd/vela-server/secret.go [hunk 1]
```diff
@@ -51,5 +51,5 @@ func setupNative(c *cli.Context, d database.Service) (secret.Service, error) {
 // helper function to setup the Vault secret engine from the CLI arguments.
 func setupVault(c *cli.Context) (secret.Service, error) {
 	logrus.Tracef("Creating %s secret client from CLI configuration", constants.DriverVault)
-	return vault.New(c.String("vault-addr"), c.String("vault-token"), c.String("vault-version"), c.String("vault-prefix"))
+	return vault.New(c.String("vault-addr"), c.String("vault-token"), c.String("vault-version"), c.String("vault-prefix"), c.String("vault-auth-method"), c.String("vault-aws-role"), c.Duration("vault-renewal"))
```
Comment (kneal) [inline cmd/vela-server/secret.go:54]: If we can think of a way to pass a struct vs values I think that would be more ideal. 

Values are good, just from a code readability standpoint, it feels better to pass a struct when you get into instances where functions have more than 4-5 parameters.

**Review by kneal — APPROVED** [review]

File: cmd/vela-server/main.go [hunk 2]
```diff
@@ -157,6 +157,22 @@ func main() {
...
+			Name:    "vault-auth-method",
+			Usage:   "auth method to utilize to obtain token",
+		},
+		&cli.StringFlag{
+			EnvVars: []string{"VELA_SECRET_VAULT_AWS_ROLE", "SECRET_VAULT_AWS_ROLE"},
+			Name:    "vault-aws-role",
+			Usage:   "vault role to connect to the auth/aws/login endpoint with",
+		},
+		&cli.DurationFlag{
+			EnvVars: []string{"VELA_SECRET_VAULT_RENEWAL", "VELA_SECRET_VAULT_RENEWAL"},
```
Comment (jbrockopp) [inline cmd/vela-server/main.go:171]: Should these be the same?

Based off your pattern above, I'm thinking it should look like:

```suggestion
			EnvVars: []string{"VELA_SECRET_VAULT_RENEWAL", "SECRET_VAULT_RENEWAL"},
```

File: cmd/vela-server/secret.go (hunk 1 above)
Comment (jbrockopp) [reply cmd/vela-server/secret.go:54]: It'd be great to come up with a standard for this in the Vela codebase 👍 

Something like:

> If a function takes more than 3 parameters of the same type, we should make it accept a struct instead.

And maybe we don't even care about the types of the parameters, and focus on the sheer number of parameters?

**Review by jbrockopp — CHANGES_REQUESTED** [review]

File: cmd/vela-server/secret.go (hunk 1 above)
Comment (JordanSussman) [reply cmd/vela-server/secret.go]: Update to utilize a `vault.Config` struct

File: cmd/vela-server/main.go (hunk 2 above)
Comment (JordanSussman) [reply cmd/vela-server/main.go]: updated

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by wass3r — APPROVED** [review]

**Comment by wass3r** [comment]
not sure what's going with the `checks`, but the last commit showed all green.. merging.

**Pull request merged**